lib/Job/Async/Worker/Redis.pm

```diff
@@ -252,6 +253,11 @@ sub trigger {
                 })->on_fail(sub {
                     my $failure = shift;
                     $log->errorf("Failed to retrieve job from redis: %s", $failure);
+                    delete $self->{awaiting_job};
```

Comment:
It is not necessary to worry about it. `$self->{awaiting_job} is not any actual queue job, but `a future awaiting for new jobs to come`. Created in line 230 above, it's purpose is just to prevent a worker from running `brpoplpush` more than once. If it is not deleted, after any failure in fetching a new job, worker will stop accepting any other job.